prprocessor/__main__.py

```diff
@@ -282,13 +320,65 @@ async def update_redmine_on_issues(pull_request: Mapping, issues: Iterable[Issue
 
 @process_event_actions('pull_request', {'opened', 'ready_for_review', 'reopened', 'synchronize'})
 @process_webhook_payload
-async def on_pr_modified(*, pull_request: Mapping, **other) -> None:  # pylint: disable=unused-argument
-    await run_pull_request_check(pull_request)
+async def on_pr_modified(*, action: str, pull_request: Mapping, **_kw) -> None:
+    commits_valid_style = await run_pull_request_check(pull_request)
+
+    original_labels: set[str] = {label['name'] for label in pull_request['labels']}
+    labels_before = Label.load_labels(original_labels)
+    labels = labels_before.copy()
+
+    if action == 'opened':
+        labels.add(Label.NOT_YET_REVIEWED)
+
+    if action == 'synchronize' and Label.WAITING_ON_CONTRIBUTOR in labels:
+        labels.remove(Label.WAITING_ON_CONTRIBUTOR)
+        if pull_request['review_comments'] > 0:
+            labels.add(Label.NEEDS_RE_REVIEW)
+
+    if not commits_valid_style:
+        labels.add(Label.WAITING_ON_CONTRIBUTOR)
+
+    # TODO: handle None value (result is being calculated by GH) and resubmit later?
+    if pull_request['mergable'] is False:
+        # TODO: post message the PR has a conflict?
+        labels.remove(Label.NEEDS_RE_REVIEW)
+        labels.remove(Label.NOT_YET_REVIEWED)
+        labels.add(Label.WAITING_ON_CONTRIBUTOR)
+
+    labels_to_add = labels - labels_before
+    labels_to_remove = labels_before - labels
+
+    # TODO: branch labels
```

Comment:
I'm unsure how to best approach this. I guess labels_to_add needs to become a set of strings. Right now I'm tempted to first implement the current logic, deploy it and make sure it works. Then port over branch labels later.